// @abp/ng.tenant-management/fesm2022/abp-ng.tenant-management-config.mjs
import { RoutesService } from '@abp/ng.core';
import * as i0 from '@angular/core';
import { provideAppInitializer, inject, makeEnvironmentProviders, NgModule } from '@angular/core';

const TENANT_MANAGEMENT_ROUTE_PROVIDERS = [
    provideAppInitializer(() => {
        configureRoutes();
    }),
];
function configureRoutes() {
    const routes = inject(RoutesService);
    routes.add([
        {
            path: undefined,
            name: "AbpTenantManagement::Menu:TenantManagement" /* eTenantManagementRouteNames.TenantManagement */,
            parentName: "AbpUiNavigation::Menu:Administration" /* eThemeSharedRouteNames.Administration */,
            requiredPolicy: "AbpTenantManagement.Tenants" /* eTenantManagementPolicyNames.TenantManagement */,
            layout: "application" /* eLayoutType.application */,
            iconClass: 'fa fa-users',
            order: 2,
        },
        {
            path: '/tenant-management/tenants',
            name: "AbpTenantManagement::Tenants" /* eTenantManagementRouteNames.Tenants */,
            parentName: "AbpTenantManagement::Menu:TenantManagement" /* eTenantManagementRouteNames.TenantManagement */,
            requiredPolicy: "AbpTenantManagement.Tenants" /* eTenantManagementPolicyNames.Tenants */,
            order: 1,
        },
    ]);
}

function provideTenantManagementConfig() {
    return makeEnvironmentProviders([TENANT_MANAGEMENT_ROUTE_PROVIDERS]);
}

/**
 * @deprecated TenantManagementConfigModule is deprecated use `provideTenantManagementConfig` *function* instead.
 */
class TenantManagementConfigModule {
    static forRoot() {
        return {
            ngModule: TenantManagementConfigModule,
            providers: [provideTenantManagementConfig()],
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: TenantManagementConfigModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "19.1.8", ngImport: i0, type: TenantManagementConfigModule }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: TenantManagementConfigModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: TenantManagementConfigModule, decorators: [{
            type: NgModule
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { TENANT_MANAGEMENT_ROUTE_PROVIDERS, TenantManagementConfigModule, configureRoutes, provideTenantManagementConfig };
