// @abp/ng.account/fesm2022/abp-ng.account.mjs
import * as i3$1 from '@abp/ng.core';
import { AuthService, ConfigStateService, InternalStore, ReplaceableRouteContainerComponent, authGuard, RouterOutletComponent, LazyModuleFactory, CoreModule } from '@abp/ng.core';
import * as i4 from '@abp/ng.theme.shared';
import { ToasterService, getPasswordValidators, ConfirmationService, Confirmation, fadeIn, ThemeSharedModule } from '@abp/ng.theme.shared';
import * as i0 from '@angular/core';
import { Component, InjectionToken, inject, Injector, Injectable, Inject, NgModule } from '@angular/core';
import { NgbDropdownModule } from '@ng-bootstrap/ng-bootstrap';
import * as i6 from '@ngx-validate/core';
import { comparePasswords, NgxValidateCoreModule } from '@ngx-validate/core';
import * as i3 from '@angular/router';
import { ActivatedRoute, RouterModule } from '@angular/router';
import * as i2 from '@abp/ng.account.core/proxy';
import { ProfileService } from '@abp/ng.account.core/proxy';
import * as i1 from '@angular/forms';
import { Validators, UntypedFormBuilder, UntypedFormControl } from '@angular/forms';
import { finalize, catchError, filter, switchMap, map, tap } from 'rxjs/operators';
import { throwError, map as map$1, tap as tap$1 } from 'rxjs';
import { trigger, transition, useAnimation } from '@angular/animations';
import * as i3$2 from '@angular/common';
import * as i4$1 from '@abp/ng.components/extensible';
import { EXTENSIONS_FORM_PROP, EXTENSIBLE_FORM_VIEW_PROVIDER, FormProp, FormPropData, generateFormFromProps, EXTENSIONS_IDENTIFIER, ExtensionsService, getObjectExtensionEntitiesFromStore, mapEntitiesToContributors, mergeWithDefaultProps, ExtensibleModule } from '@abp/ng.components/extensible';

class ForgotPasswordComponent {
    constructor(fb, accountService) {
        this.fb = fb;
        this.accountService = accountService;
        this.isEmailSent = false;
        this.form = this.fb.group({
            email: ['', [Validators.required, Validators.email]],
        });
    }
    onSubmit() {
        if (this.form.invalid)
            return;
        this.inProgress = true;
        this.accountService
            .sendPasswordResetCode({
            email: this.form.get('email')?.value,
            appName: 'Angular',
        })
            .pipe(finalize(() => (this.inProgress = false)))
            .subscribe(() => {
            this.isEmailSent = true;
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ForgotPasswordComponent, deps: [{ token: i1.UntypedFormBuilder }, { token: i2.AccountService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "19.1.8", type: ForgotPasswordComponent, isStandalone: false, selector: "abp-forgot-password", ngImport: i0, template: "<h4>{{ 'AbpAccount::ForgotPassword' | abpLocalization }}</h4>\r\n\r\n@if (!isEmailSent) {\r\n  <form [formGroup]=\"form\" (ngSubmit)=\"onSubmit()\" validateOnSubmit>\r\n    <p>{{ 'AbpAccount::SendPasswordResetLink_Information' | abpLocalization }}</p>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-email-address\" class=\"form-label\">{{\r\n        'AbpAccount::EmailAddress' | abpLocalization\r\n      }}</label\r\n      ><span> * </span>\r\n      <input type=\"email\" id=\"input-email-address\" class=\"form-control\" formControlName=\"email\" />\r\n    </div>\r\n    <abp-button\r\n      class=\"d-block\"\r\n      buttonClass=\"mt-2 mb-3 btn btn-primary btn-block\"\r\n      [loading]=\"inProgress\"\r\n      buttonType=\"submit\"\r\n      [disabled]=\"form?.invalid\"\r\n    >\r\n      {{ 'AbpAccount::Submit' | abpLocalization }}\r\n    </abp-button>\r\n    <a routerLink=\"/account/login\"\r\n      ><i class=\"fa fa-long-arrow-left me-1\" aria-hidden=\"true\"></i\r\n      >{{ 'AbpAccount::Login' | abpLocalization }}</a\r\n    >\r\n  </form>\r\n} @else {\r\n  <p>\r\n    {{ 'AbpAccount::PasswordResetMailSentMessage' | abpLocalization }}\r\n  </p>\r\n\r\n  <a routerLink=\"/account/login\">\r\n    <button class=\"d-block mt-2 mb-3 btn btn-primary btn-block\">\r\n      <i class=\"fa fa-long-arrow-left me-1\" aria-hidden=\"true\"></i>\r\n      {{ 'AbpAccount::BackToLogin' | abpLocalization }}\r\n    </button>\r\n  </a>\r\n}\r\n", dependencies: [{ kind: "directive", type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { kind: "directive", type: i1.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { kind: "directive", type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { kind: "directive", type: i1.FormControlName, selector: "[formControlName]", inputs: ["formControlName", "disabled", "ngModel"], outputs: ["ngModelChange"] }, { kind: "directive", type: i3.RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "info", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "directive", type: i3$1.FormSubmitDirective, selector: "form[ngSubmit][formGroup]", inputs: ["debounce", "notValidateOnSubmit", "markAsDirtyWhenSubmit"], outputs: ["ngSubmit"] }, { kind: "directive", type: i6.ValidationGroupDirective, selector: "[formGroup],[formGroupName]", exportAs: ["validationGroup"] }, { kind: "directive", type: i6.ValidationDirective, selector: "[formControl],[formControlName]", exportAs: ["validationDirective"] }, { kind: "component", type: i4.ButtonComponent, selector: "abp-button", inputs: ["buttonId", "buttonClass", "buttonType", "formName", "iconClass", "loading", "disabled", "attributes"], outputs: ["click", "focus", "blur", "abpClick", "abpFocus", "abpBlur"] }, { kind: "pipe", type: i3$1.LocalizationPipe, name: "abpLocalization" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ForgotPasswordComponent, decorators: [{
            type: Component,
            args: [{ standalone: false, selector: 'abp-forgot-password', template: "<h4>{{ 'AbpAccount::ForgotPassword' | abpLocalization }}</h4>\r\n\r\n@if (!isEmailSent) {\r\n  <form [formGroup]=\"form\" (ngSubmit)=\"onSubmit()\" validateOnSubmit>\r\n    <p>{{ 'AbpAccount::SendPasswordResetLink_Information' | abpLocalization }}</p>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-email-address\" class=\"form-label\">{{\r\n        'AbpAccount::EmailAddress' | abpLocalization\r\n      }}</label\r\n      ><span> * </span>\r\n      <input type=\"email\" id=\"input-email-address\" class=\"form-control\" formControlName=\"email\" />\r\n    </div>\r\n    <abp-button\r\n      class=\"d-block\"\r\n      buttonClass=\"mt-2 mb-3 btn btn-primary btn-block\"\r\n      [loading]=\"inProgress\"\r\n      buttonType=\"submit\"\r\n      [disabled]=\"form?.invalid\"\r\n    >\r\n      {{ 'AbpAccount::Submit' | abpLocalization }}\r\n    </abp-button>\r\n    <a routerLink=\"/account/login\"\r\n      ><i class=\"fa fa-long-arrow-left me-1\" aria-hidden=\"true\"></i\r\n      >{{ 'AbpAccount::Login' | abpLocalization }}</a\r\n    >\r\n  </form>\r\n} @else {\r\n  <p>\r\n    {{ 'AbpAccount::PasswordResetMailSentMessage' | abpLocalization }}\r\n  </p>\r\n\r\n  <a routerLink=\"/account/login\">\r\n    <button class=\"d-block mt-2 mb-3 btn btn-primary btn-block\">\r\n      <i class=\"fa fa-long-arrow-left me-1\" aria-hidden=\"true\"></i>\r\n      {{ 'AbpAccount::BackToLogin' | abpLocalization }}\r\n    </button>\r\n  </a>\r\n}\r\n" }]
        }], ctorParameters: () => [{ type: i1.UntypedFormBuilder }, { type: i2.AccountService }] });

const ACCOUNT_CONFIG_OPTIONS = new InjectionToken('ACCOUNT_CONFIG_OPTIONS');

function getRedirectUrl(injector) {
    const route = injector.get(ActivatedRoute);
    const options = injector.get(ACCOUNT_CONFIG_OPTIONS);
    return route.snapshot.queryParams.returnUrl || options.redirectUrl || '/';
}

const { maxLength: maxLength$2, required: required$3 } = Validators;
class LoginComponent {
    constructor() {
        this.injector = inject(Injector);
        this.fb = inject(UntypedFormBuilder);
        this.toasterService = inject(ToasterService);
        this.authService = inject(AuthService);
        this.configState = inject(ConfigStateService);
        this.isSelfRegistrationEnabled = true;
        this.authWrapperKey = "Account.AuthWrapperComponent" /* eAccountComponents.AuthWrapper */;
    }
    ngOnInit() {
        this.init();
        this.buildForm();
    }
    init() {
        this.isSelfRegistrationEnabled =
            (this.configState.getSetting('Abp.Account.IsSelfRegistrationEnabled') || '').toLowerCase() !== 'false';
    }
    buildForm() {
        this.form = this.fb.group({
            username: ['', [required$3, maxLength$2(255)]],
            password: ['', [required$3, maxLength$2(128)]],
            rememberMe: [false],
        });
    }
    onSubmit() {
        if (this.form.invalid)
            return;
        this.inProgress = true;
        const { username, password, rememberMe } = this.form.value;
        const redirectUrl = getRedirectUrl(this.injector);
        this.authService
            .login({ username, password, rememberMe, redirectUrl })
            .pipe(catchError(err => {
            this.toasterService.error(err.error?.error_description ||
                err.error?.error.message ||
                'AbpAccount::DefaultErrorMessage', '', { life: 7000 });
            return throwError(err);
        }), finalize(() => (this.inProgress = false)))
            .subscribe();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: LoginComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "19.1.8", type: LoginComponent, isStandalone: false, selector: "abp-login", ngImport: i0, template: "<h4>{{ 'AbpAccount::Login' | abpLocalization }}</h4>\r\n@if (isSelfRegistrationEnabled) {\r\n  <strong>\r\n    {{ 'AbpAccount::AreYouANewUser' | abpLocalization }}\r\n    <a class=\"text-decoration-none\" routerLink=\"/account/register\" queryParamsHandling=\"preserve\">{{\r\n      'AbpAccount::Register' | abpLocalization\r\n    }}</a>\r\n  </strong>\r\n}\r\n<form [formGroup]=\"form\" (ngSubmit)=\"onSubmit()\" validateOnSubmit class=\"mt-4\">\r\n  <div class=\"mb-3 form-group\">\r\n    <label for=\"login-input-user-name-or-email-address\" class=\"form-label\">{{\r\n      'AbpAccount::UserNameOrEmailAddress' | abpLocalization\r\n    }}</label>\r\n    <input\r\n      class=\"form-control\"\r\n      type=\"text\"\r\n      id=\"login-input-user-name-or-email-address\"\r\n      formControlName=\"username\"\r\n      autocomplete=\"username\"\r\n      autofocus\r\n    />\r\n  </div>\r\n  <div class=\"mb-3 form-group\">\r\n    <label for=\"login-input-password\" class=\"form-label\">{{\r\n      'AbpAccount::Password' | abpLocalization\r\n    }}</label>\r\n    <input\r\n      class=\"form-control\"\r\n      type=\"password\"\r\n      id=\"login-input-password\"\r\n      formControlName=\"password\"\r\n      autocomplete=\"current-password\"\r\n    />\r\n  </div>\r\n\r\n  <div class=\"row\">\r\n    <div class=\"col\">\r\n      <div class=\"form-check\">\r\n        <label class=\"form-check-label mb-2\" for=\"login-input-remember-me\">\r\n          <input\r\n            class=\"form-check-input\"\r\n            type=\"checkbox\"\r\n            id=\"login-input-remember-me\"\r\n            formControlName=\"rememberMe\"\r\n          />\r\n          {{ 'AbpAccount::RememberMe' | abpLocalization }}\r\n        </label>\r\n      </div>\r\n    </div>\r\n    <div class=\"text-end col\">\r\n      <a routerLink=\"/account/forgot-password\">{{\r\n        'AbpAccount::ForgotPassword' | abpLocalization\r\n      }}</a>\r\n    </div>\r\n  </div>\r\n\r\n  <abp-button\r\n    [loading]=\"inProgress\"\r\n    buttonType=\"submit\"\r\n    name=\"Action\"\r\n    buttonClass=\"btn-block btn-lg mt-3 btn btn-primary\"\r\n  >\r\n    {{ 'AbpAccount::Login' | abpLocalization }}\r\n  </abp-button>\r\n</form>\r\n", dependencies: [{ kind: "directive", type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { kind: "directive", type: i1.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i1.CheckboxControlValueAccessor, selector: "input[type=checkbox][formControlName],input[type=checkbox][formControl],input[type=checkbox][ngModel]" }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { kind: "directive", type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { kind: "directive", type: i1.FormControlName, selector: "[formControlName]", inputs: ["formControlName", "disabled", "ngModel"], outputs: ["ngModelChange"] }, { kind: "directive", type: i3.RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "info", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "directive", type: i3$1.AutofocusDirective, selector: "[autofocus]", inputs: ["autofocus"] }, { kind: "directive", type: i3$1.FormSubmitDirective, selector: "form[ngSubmit][formGroup]", inputs: ["debounce", "notValidateOnSubmit", "markAsDirtyWhenSubmit"], outputs: ["ngSubmit"] }, { kind: "directive", type: i6.ValidationGroupDirective, selector: "[formGroup],[formGroupName]", exportAs: ["validationGroup"] }, { kind: "directive", type: i6.ValidationDirective, selector: "[formControl],[formControlName]", exportAs: ["validationDirective"] }, { kind: "component", type: i4.ButtonComponent, selector: "abp-button", inputs: ["buttonId", "buttonClass", "buttonType", "formName", "iconClass", "loading", "disabled", "attributes"], outputs: ["click", "focus", "blur", "abpClick", "abpFocus", "abpBlur"] }, { kind: "pipe", type: i3$1.LocalizationPipe, name: "abpLocalization" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: LoginComponent, decorators: [{
            type: Component,
            args: [{ standalone: false, selector: 'abp-login', template: "<h4>{{ 'AbpAccount::Login' | abpLocalization }}</h4>\r\n@if (isSelfRegistrationEnabled) {\r\n  <strong>\r\n    {{ 'AbpAccount::AreYouANewUser' | abpLocalization }}\r\n    <a class=\"text-decoration-none\" routerLink=\"/account/register\" queryParamsHandling=\"preserve\">{{\r\n      'AbpAccount::Register' | abpLocalization\r\n    }}</a>\r\n  </strong>\r\n}\r\n<form [formGroup]=\"form\" (ngSubmit)=\"onSubmit()\" validateOnSubmit class=\"mt-4\">\r\n  <div class=\"mb-3 form-group\">\r\n    <label for=\"login-input-user-name-or-email-address\" class=\"form-label\">{{\r\n      'AbpAccount::UserNameOrEmailAddress' | abpLocalization\r\n    }}</label>\r\n    <input\r\n      class=\"form-control\"\r\n      type=\"text\"\r\n      id=\"login-input-user-name-or-email-address\"\r\n      formControlName=\"username\"\r\n      autocomplete=\"username\"\r\n      autofocus\r\n    />\r\n  </div>\r\n  <div class=\"mb-3 form-group\">\r\n    <label for=\"login-input-password\" class=\"form-label\">{{\r\n      'AbpAccount::Password' | abpLocalization\r\n    }}</label>\r\n    <input\r\n      class=\"form-control\"\r\n      type=\"password\"\r\n      id=\"login-input-password\"\r\n      formControlName=\"password\"\r\n      autocomplete=\"current-password\"\r\n    />\r\n  </div>\r\n\r\n  <div class=\"row\">\r\n    <div class=\"col\">\r\n      <div class=\"form-check\">\r\n        <label class=\"form-check-label mb-2\" for=\"login-input-remember-me\">\r\n          <input\r\n            class=\"form-check-input\"\r\n            type=\"checkbox\"\r\n            id=\"login-input-remember-me\"\r\n            formControlName=\"rememberMe\"\r\n          />\r\n          {{ 'AbpAccount::RememberMe' | abpLocalization }}\r\n        </label>\r\n      </div>\r\n    </div>\r\n    <div class=\"text-end col\">\r\n      <a routerLink=\"/account/forgot-password\">{{\r\n        'AbpAccount::ForgotPassword' | abpLocalization\r\n      }}</a>\r\n    </div>\r\n  </div>\r\n\r\n  <abp-button\r\n    [loading]=\"inProgress\"\r\n    buttonType=\"submit\"\r\n    name=\"Action\"\r\n    buttonClass=\"btn-block btn-lg mt-3 btn btn-primary\"\r\n  >\r\n    {{ 'AbpAccount::Login' | abpLocalization }}\r\n  </abp-button>\r\n</form>\r\n" }]
        }] });

class ManageProfileStateService {
    constructor() {
        this.store = new InternalStore({});
    }
    get createOnUpdateStream() {
        return this.store.sliceUpdate;
    }
    getProfile$() {
        return this.store.sliceState(state => state.profile);
    }
    getProfile() {
        return this.store.state.profile;
    }
    setProfile(profile) {
        this.store.patch({ profile });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ManageProfileStateService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ManageProfileStateService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ManageProfileStateService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }] });

const { required: required$2 } = Validators;
const PASSWORD_FIELDS$1 = ['newPassword', 'repeatNewPassword'];
class ChangePasswordComponent {
    constructor(fb, injector, toasterService, profileService, manageProfileState) {
        this.fb = fb;
        this.injector = injector;
        this.toasterService = toasterService;
        this.profileService = profileService;
        this.manageProfileState = manageProfileState;
        this.mapErrorsFn = (errors, groupErrors, control) => {
            if (PASSWORD_FIELDS$1.indexOf(String(control?.name)) < 0)
                return errors;
            return errors.concat(groupErrors.filter(({ key }) => key === 'passwordMismatch'));
        };
    }
    ngOnInit() {
        this.hideCurrentPassword = !this.manageProfileState.getProfile()?.hasPassword;
        const passwordValidations = getPasswordValidators(this.injector);
        this.form = this.fb.group({
            password: ['', required$2],
            newPassword: [
                '',
                {
                    validators: [required$2, ...passwordValidations],
                },
            ],
            repeatNewPassword: [
                '',
                {
                    validators: [required$2, ...passwordValidations],
                },
            ],
        }, {
            validators: [comparePasswords(PASSWORD_FIELDS$1)],
        });
        if (this.hideCurrentPassword)
            this.form.removeControl('password');
    }
    onSubmit() {
        if (this.form.invalid)
            return;
        this.inProgress = true;
        this.profileService
            .changePassword({
            ...(!this.hideCurrentPassword && { currentPassword: this.form.get('password')?.value }),
            newPassword: this.form.get('newPassword')?.value,
        })
            .pipe(finalize(() => (this.inProgress = false)))
            .subscribe({
            next: () => {
                this.form.reset();
                this.toasterService.success('AbpAccount::PasswordChangedMessage', '', {
                    life: 5000,
                });
                if (this.hideCurrentPassword) {
                    this.hideCurrentPassword = false;
                    this.form.addControl('password', new UntypedFormControl('', [required$2]));
                }
            },
            error: err => {
                this.toasterService.error(err.error?.error?.message || 'AbpAccount::DefaultErrorMessage');
            },
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ChangePasswordComponent, deps: [{ token: i1.UntypedFormBuilder }, { token: i0.Injector }, { token: i4.ToasterService }, { token: i2.ProfileService }, { token: ManageProfileStateService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "19.1.8", type: ChangePasswordComponent, isStandalone: false, selector: "abp-change-password-form", exportAs: ["abpChangePasswordForm"], ngImport: i0, template: "<form [formGroup]=\"form\" (ngSubmit)=\"onSubmit()\" [mapErrorsFn]=\"mapErrorsFn\" validateOnSubmit class=\"abp-md-form\">\r\n  @if (!hideCurrentPassword) {\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"current-password\" class=\"form-label\">{{\r\n        'AbpIdentity::DisplayName:CurrentPassword' | abpLocalization\r\n      }}</label\r\n      ><span> * </span\r\n      ><input\r\n        type=\"password\"\r\n        id=\"current-password\"\r\n        class=\"form-control\"\r\n        formControlName=\"password\"\r\n        autofocus\r\n        autocomplete=\"current-password\"\r\n      />\r\n    </div>\r\n  }\r\n  <div class=\"mb-3 form-group\">\r\n    <label for=\"new-password\" class=\"form-label\">{{\r\n      'AbpIdentity::DisplayName:NewPassword' | abpLocalization\r\n    }}</label\r\n    ><span> * </span\r\n    ><input\r\n      type=\"password\"\r\n      id=\"new-password\"\r\n      class=\"form-control\"\r\n      formControlName=\"newPassword\"\r\n      autocomplete=\"new-password\"\r\n    />\r\n  </div>\r\n  <div class=\"mb-3 form-group\">\r\n    <label for=\"confirm-new-password\" class=\"form-label\">{{\r\n      'AbpIdentity::DisplayName:NewPasswordConfirm' | abpLocalization\r\n    }}</label\r\n    ><span> * </span\r\n    ><input\r\n      type=\"password\"\r\n      id=\"confirm-new-password\"\r\n      class=\"form-control\"\r\n      formControlName=\"repeatNewPassword\"\r\n      autocomplete=\"new-password\"\r\n    />\r\n  </div>\r\n  <abp-button\r\n    iconClass=\"fa fa-check\"\r\n    buttonClass=\"btn btn-primary color-white\"\r\n    buttonType=\"submit\"\r\n    [loading]=\"inProgress\"\r\n    [disabled]=\"form?.invalid\"\r\n    >{{ 'AbpIdentity::Save' | abpLocalization }}</abp-button\r\n  >\r\n</form>\r\n", dependencies: [{ kind: "directive", type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { kind: "directive", type: i1.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { kind: "directive", type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { kind: "directive", type: i1.FormControlName, selector: "[formControlName]", inputs: ["formControlName", "disabled", "ngModel"], outputs: ["ngModelChange"] }, { kind: "directive", type: i3$1.AutofocusDirective, selector: "[autofocus]", inputs: ["autofocus"] }, { kind: "directive", type: i3$1.FormSubmitDirective, selector: "form[ngSubmit][formGroup]", inputs: ["debounce", "notValidateOnSubmit", "markAsDirtyWhenSubmit"], outputs: ["ngSubmit"] }, { kind: "directive", type: i6.ValidationGroupDirective, selector: "[formGroup],[formGroupName]", exportAs: ["validationGroup"] }, { kind: "directive", type: i6.ValidationDirective, selector: "[formControl],[formControlName]", exportAs: ["validationDirective"] }, { kind: "component", type: i4.ButtonComponent, selector: "abp-button", inputs: ["buttonId", "buttonClass", "buttonType", "formName", "iconClass", "loading", "disabled", "attributes"], outputs: ["click", "focus", "blur", "abpClick", "abpFocus", "abpBlur"] }, { kind: "pipe", type: i3$1.LocalizationPipe, name: "abpLocalization" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ChangePasswordComponent, decorators: [{
            type: Component,
            args: [{ standalone: false, selector: 'abp-change-password-form', exportAs: 'abpChangePasswordForm', template: "<form [formGroup]=\"form\" (ngSubmit)=\"onSubmit()\" [mapErrorsFn]=\"mapErrorsFn\" validateOnSubmit class=\"abp-md-form\">\r\n  @if (!hideCurrentPassword) {\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"current-password\" class=\"form-label\">{{\r\n        'AbpIdentity::DisplayName:CurrentPassword' | abpLocalization\r\n      }}</label\r\n      ><span> * </span\r\n      ><input\r\n        type=\"password\"\r\n        id=\"current-password\"\r\n        class=\"form-control\"\r\n        formControlName=\"password\"\r\n        autofocus\r\n        autocomplete=\"current-password\"\r\n      />\r\n    </div>\r\n  }\r\n  <div class=\"mb-3 form-group\">\r\n    <label for=\"new-password\" class=\"form-label\">{{\r\n      'AbpIdentity::DisplayName:NewPassword' | abpLocalization\r\n    }}</label\r\n    ><span> * </span\r\n    ><input\r\n      type=\"password\"\r\n      id=\"new-password\"\r\n      class=\"form-control\"\r\n      formControlName=\"newPassword\"\r\n      autocomplete=\"new-password\"\r\n    />\r\n  </div>\r\n  <div class=\"mb-3 form-group\">\r\n    <label for=\"confirm-new-password\" class=\"form-label\">{{\r\n      'AbpIdentity::DisplayName:NewPasswordConfirm' | abpLocalization\r\n    }}</label\r\n    ><span> * </span\r\n    ><input\r\n      type=\"password\"\r\n      id=\"confirm-new-password\"\r\n      class=\"form-control\"\r\n      formControlName=\"repeatNewPassword\"\r\n      autocomplete=\"new-password\"\r\n    />\r\n  </div>\r\n  <abp-button\r\n    iconClass=\"fa fa-check\"\r\n    buttonClass=\"btn btn-primary color-white\"\r\n    buttonType=\"submit\"\r\n    [loading]=\"inProgress\"\r\n    [disabled]=\"form?.invalid\"\r\n    >{{ 'AbpIdentity::Save' | abpLocalization }}</abp-button\r\n  >\r\n</form>\r\n" }]
        }], ctorParameters: () => [{ type: i1.UntypedFormBuilder }, { type: i0.Injector }, { type: i4.ToasterService }, { type: i2.ProfileService }, { type: ManageProfileStateService }] });

const RE_LOGIN_CONFIRMATION_TOKEN = new InjectionToken('RE_LOGIN_CONFIRMATION_TOKEN');

class PersonalSettingsHalfRowComponent {
    constructor(propData) {
        this.propData = propData;
        this.displayName = propData.displayName;
        this.name = propData.name;
        this.id = propData.id || '';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: PersonalSettingsHalfRowComponent, deps: [{ token: EXTENSIONS_FORM_PROP }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "19.1.8", type: PersonalSettingsHalfRowComponent, isStandalone: false, selector: "abp-personal-settings-half-row", ngImport: i0, template: ` <div class="w-50 d-inline">
    <label [attr.for]="name" class="form-label">{{ displayName | abpLocalization }} </label>
    <input
      type="text"
      [attr.id]="id"
      class="form-control"
      [attr.name]="name"
      [formControlName]="name"
    />
  </div>`, isInline: true, dependencies: [{ kind: "directive", type: i1.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.FormControlName, selector: "[formControlName]", inputs: ["formControlName", "disabled", "ngModel"], outputs: ["ngModelChange"] }, { kind: "directive", type: i6.ValidationDirective, selector: "[formControl],[formControlName]", exportAs: ["validationDirective"] }, { kind: "pipe", type: i3$1.LocalizationPipe, name: "abpLocalization" }], viewProviders: [EXTENSIBLE_FORM_VIEW_PROVIDER] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: PersonalSettingsHalfRowComponent, decorators: [{
            type: Component,
            args: [{ standalone: false, selector: 'abp-personal-settings-half-row', template: ` <div class="w-50 d-inline">
    <label [attr.for]="name" class="form-label">{{ displayName | abpLocalization }} </label>
    <input
      type="text"
      [attr.id]="id"
      class="form-control"
      [attr.name]="name"
      [formControlName]="name"
    />
  </div>`, viewProviders: [EXTENSIBLE_FORM_VIEW_PROVIDER] }]
        }], ctorParameters: () => [{ type: i4$1.FormProp, decorators: [{
                    type: Inject,
                    args: [EXTENSIONS_FORM_PROP]
                }] }] });

const { maxLength: maxLength$1, required: required$1, email: email$1 } = Validators;
const DEFAULT_PERSONAL_SETTINGS_UPDATE_FORM_PROPS = FormProp.createMany([
    {
        type: "string" /* ePropType.String */,
        name: 'userName',
        displayName: 'AbpIdentity::DisplayName:UserName',
        id: 'username',
        validators: () => [required$1, maxLength$1(256)],
    },
    {
        type: "string" /* ePropType.String */,
        name: 'name',
        displayName: 'AbpIdentity::DisplayName:Name',
        id: 'name',
        validators: () => [maxLength$1(64)],
        template: PersonalSettingsHalfRowComponent,
        className: 'd-inline-block w-50',
    },
    {
        type: "string" /* ePropType.String */,
        name: 'surname',
        displayName: 'AbpIdentity::DisplayName:Surname',
        id: 'surname',
        validators: () => [maxLength$1(64)],
        className: 'd-inline-block w-50 ps-4',
        template: PersonalSettingsHalfRowComponent,
    },
    {
        type: "string" /* ePropType.String */,
        name: 'email',
        displayName: 'AbpIdentity::DisplayName:Email',
        id: 'email-address',
        validators: () => [required$1, email$1, maxLength$1(256)],
    },
    {
        type: "string" /* ePropType.String */,
        name: 'phoneNumber',
        displayName: 'AbpIdentity::DisplayName:PhoneNumber',
        id: 'phone-number',
        validators: () => [maxLength$1(16)],
    },
]);

const DEFAULT_ACCOUNT_FORM_PROPS = {
    ["Account.PersonalSettingsComponent" /* eAccountComponents.PersonalSettings */]: DEFAULT_PERSONAL_SETTINGS_UPDATE_FORM_PROPS,
};
const ACCOUNT_EDIT_FORM_PROP_CONTRIBUTORS = new InjectionToken('ACCOUNT_EDIT_FORM_PROP_CONTRIBUTORS');

class PersonalSettingsComponent {
    constructor() {
        this.fb = inject(UntypedFormBuilder);
        this.toasterService = inject(ToasterService);
        this.profileService = inject(ProfileService);
        this.manageProfileState = inject(ManageProfileStateService);
        this.authService = inject(AuthService);
        this.confirmationService = inject(ConfirmationService);
        this.configState = inject(ConfigStateService);
        this.isPersonalSettingsChangedConfirmationActive = inject(RE_LOGIN_CONFIRMATION_TOKEN);
        this.injector = inject(Injector);
        this.logoutConfirmation = () => {
            this.authService.logout().subscribe();
        };
    }
    buildForm() {
        this.selected = this.manageProfileState.getProfile();
        if (!this.selected) {
            return;
        }
        const data = new FormPropData(this.injector, this.selected);
        this.form = generateFormFromProps(data);
    }
    ngOnInit() {
        this.buildForm();
    }
    submit() {
        if (this.form.invalid)
            return;
        const isLogOutConfirmMessageVisible = this.isLogoutConfirmMessageActive();
        const isRefreshTokenExists = this.authService.getRefreshToken();
        this.inProgress = true;
        this.profileService
            .update(this.form.value)
            .pipe(finalize(() => (this.inProgress = false)))
            .subscribe(profile => {
            this.manageProfileState.setProfile(profile);
            this.configState.refreshAppState();
            this.toasterService.success('AbpAccount::PersonalSettingsSaved', 'Success', { life: 5000 });
            if (isRefreshTokenExists) {
                return this.authService.refreshToken();
            }
            if (isLogOutConfirmMessageVisible) {
                this.showLogoutConfirmMessage();
            }
        });
    }
    isLogoutConfirmMessageActive() {
        return this.isPersonalSettingsChangedConfirmationActive;
    }
    showLogoutConfirmMessage() {
        this.confirmationService
            .info('AbpAccount::PersonalSettingsChangedConfirmationModalDescription', 'AbpAccount::PersonalSettingsChangedConfirmationModalTitle')
            .pipe(filter(status => status === Confirmation.Status.confirm))
            .subscribe(this.logoutConfirmation);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: PersonalSettingsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "19.1.8", type: PersonalSettingsComponent, isStandalone: false, selector: "abp-personal-settings-form", providers: [
            {
                provide: EXTENSIONS_IDENTIFIER,
                useValue: "Account.PersonalSettingsComponent" /* eAccountComponents.PersonalSettings */,
            },
        ], exportAs: ["abpPersonalSettingsForm"], ngImport: i0, template: "@if (form) {\r\n  <form [formGroup]=\"form\" (ngSubmit)=\"submit()\" validateOnSubmit class=\"abp-md-form\">\r\n    <abp-extensible-form [selectedRecord]=\"selected\"></abp-extensible-form>\r\n  \r\n    <abp-button\r\n      buttonType=\"submit\"\r\n      iconClass=\"fa fa-check\"\r\n      buttonClass=\"btn btn-primary color-white\"\r\n      [loading]=\"inProgress\"\r\n    >\r\n      {{ 'AbpIdentity::Save' | abpLocalization }}</abp-button\r\n    >\r\n  </form>\r\n}\r\n", dependencies: [{ kind: "directive", type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { kind: "directive", type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { kind: "directive", type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { kind: "directive", type: i3$1.FormSubmitDirective, selector: "form[ngSubmit][formGroup]", inputs: ["debounce", "notValidateOnSubmit", "markAsDirtyWhenSubmit"], outputs: ["ngSubmit"] }, { kind: "directive", type: i6.ValidationGroupDirective, selector: "[formGroup],[formGroupName]", exportAs: ["validationGroup"] }, { kind: "component", type: i4.ButtonComponent, selector: "abp-button", inputs: ["buttonId", "buttonClass", "buttonType", "formName", "iconClass", "loading", "disabled", "attributes"], outputs: ["click", "focus", "blur", "abpClick", "abpFocus", "abpBlur"] }, { kind: "component", type: i4$1.ExtensibleFormComponent, selector: "abp-extensible-form", inputs: ["selectedRecord"], exportAs: ["abpExtensibleForm"] }, { kind: "pipe", type: i3$1.LocalizationPipe, name: "abpLocalization" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: PersonalSettingsComponent, decorators: [{
            type: Component,
            args: [{ standalone: false, selector: 'abp-personal-settings-form', exportAs: 'abpPersonalSettingsForm', providers: [
                        {
                            provide: EXTENSIONS_IDENTIFIER,
                            useValue: "Account.PersonalSettingsComponent" /* eAccountComponents.PersonalSettings */,
                        },
                    ], template: "@if (form) {\r\n  <form [formGroup]=\"form\" (ngSubmit)=\"submit()\" validateOnSubmit class=\"abp-md-form\">\r\n    <abp-extensible-form [selectedRecord]=\"selected\"></abp-extensible-form>\r\n  \r\n    <abp-button\r\n      buttonType=\"submit\"\r\n      iconClass=\"fa fa-check\"\r\n      buttonClass=\"btn btn-primary color-white\"\r\n      [loading]=\"inProgress\"\r\n    >\r\n      {{ 'AbpIdentity::Save' | abpLocalization }}</abp-button\r\n    >\r\n  </form>\r\n}\r\n" }]
        }] });

class ManageProfileComponent {
    constructor(profileService, manageProfileState) {
        this.profileService = profileService;
        this.manageProfileState = manageProfileState;
        this.selectedTab = 0;
        this.changePasswordKey = "Account.ChangePasswordComponent" /* eAccountComponents.ChangePassword */;
        this.personalSettingsKey = "Account.PersonalSettingsComponent" /* eAccountComponents.PersonalSettings */;
        this.profile$ = this.manageProfileState.getProfile$();
    }
    ngOnInit() {
        this.profileService.get().subscribe(profile => {
            this.manageProfileState.setProfile(profile);
            if (profile.isExternal) {
                this.hideChangePasswordTab = true;
                this.selectedTab = 1;
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ManageProfileComponent, deps: [{ token: i2.ProfileService }, { token: ManageProfileStateService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "19.1.8", type: ManageProfileComponent, isStandalone: false, selector: "abp-manage-profile", ngImport: i0, template: "<div id=\"AbpContentToolbar\"></div>\r\n<div class=\"card border-0 shadow-sm min-h-400\" [abpLoading]=\"!(profile$ | async)?.userName\">\r\n  <div class=\"card-body\">\r\n    <div class=\"row\">\r\n      <div class=\"col-12 col-md-3\">\r\n        <ul class=\"nav flex-column nav-pills\" id=\"nav-tab\" role=\"tablist\">\r\n          @if (!hideChangePasswordTab && (profile$ | async)) {\r\n            <li class=\"nav-item\" (click)=\"selectedTab = 0\">\r\n              <a\r\n                class=\"nav-link\"\r\n                [ngClass]=\"{ active: selectedTab === 0 }\"\r\n                role=\"tab\"\r\n                href=\"javascript:void(0)\"\r\n                >{{ 'AbpUi::ChangePassword' | abpLocalization }}</a\r\n              >\r\n            </li>\r\n          }\r\n          <li class=\"nav-item mb-2\" (click)=\"selectedTab = 1\">\r\n            <a\r\n              class=\"nav-link\"\r\n              [ngClass]=\"{ active: selectedTab === 1 }\"\r\n              role=\"tab\"\r\n              href=\"javascript:void(0)\"\r\n              >{{ 'AbpAccount::PersonalSettings' | abpLocalization }}</a\r\n            >\r\n          </li>\r\n        </ul>\r\n      </div>\r\n      @if (profile$ | async) {\r\n        <div class=\"col-12 col-md-9\">\r\n          @if (selectedTab === 0) {\r\n            <div class=\"tab-content\" [@fadeIn]>\r\n              <div class=\"tab-pane active\" role=\"tabpanel\">\r\n                <h4>\r\n                  {{ 'AbpIdentity::ChangePassword' | abpLocalization }}\r\n                  <hr />\r\n                </h4>\r\n                <abp-change-password-form\r\n                  *abpReplaceableTemplate=\"{\r\n                    componentKey: changePasswordKey\r\n                  }\"\r\n                ></abp-change-password-form>\r\n              </div>\r\n            </div>\r\n          }\r\n          @if (selectedTab === 1) {\r\n            <div class=\"tab-content\" [@fadeIn]>\r\n              <div class=\"tab-pane active\" role=\"tabpanel\">\r\n                <h4>\r\n                  {{ 'AbpIdentity::PersonalSettings' | abpLocalization }}\r\n                  <hr />\r\n                </h4>\r\n                <abp-personal-settings-form\r\n                  *abpReplaceableTemplate=\"{\r\n                    componentKey: personalSettingsKey\r\n                  }\"\r\n                ></abp-personal-settings-form>\r\n              </div>\r\n            </div>\r\n          }\r\n        </div>\r\n      }\r\n    </div>\r\n  </div>\r\n</div>\r\n", styles: [".min-h-400{min-height:400px}\n"], dependencies: [{ kind: "directive", type: i3$2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i3$1.ReplaceableTemplateDirective, selector: "[abpReplaceableTemplate]", inputs: ["abpReplaceableTemplate"] }, { kind: "directive", type: i4.LoadingDirective, selector: "[abpLoading]", inputs: ["abpLoading", "abpLoadingTargetElement", "abpLoadingDelay"] }, { kind: "component", type: ChangePasswordComponent, selector: "abp-change-password-form", exportAs: ["abpChangePasswordForm"] }, { kind: "component", type: PersonalSettingsComponent, selector: "abp-personal-settings-form", exportAs: ["abpPersonalSettingsForm"] }, { kind: "pipe", type: i3$2.AsyncPipe, name: "async" }, { kind: "pipe", type: i3$1.LocalizationPipe, name: "abpLocalization" }], animations: [trigger('fadeIn', [transition(':enter', useAnimation(fadeIn))])] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ManageProfileComponent, decorators: [{
            type: Component,
            args: [{ standalone: false, selector: 'abp-manage-profile', animations: [trigger('fadeIn', [transition(':enter', useAnimation(fadeIn))])], template: "<div id=\"AbpContentToolbar\"></div>\r\n<div class=\"card border-0 shadow-sm min-h-400\" [abpLoading]=\"!(profile$ | async)?.userName\">\r\n  <div class=\"card-body\">\r\n    <div class=\"row\">\r\n      <div class=\"col-12 col-md-3\">\r\n        <ul class=\"nav flex-column nav-pills\" id=\"nav-tab\" role=\"tablist\">\r\n          @if (!hideChangePasswordTab && (profile$ | async)) {\r\n            <li class=\"nav-item\" (click)=\"selectedTab = 0\">\r\n              <a\r\n                class=\"nav-link\"\r\n                [ngClass]=\"{ active: selectedTab === 0 }\"\r\n                role=\"tab\"\r\n                href=\"javascript:void(0)\"\r\n                >{{ 'AbpUi::ChangePassword' | abpLocalization }}</a\r\n              >\r\n            </li>\r\n          }\r\n          <li class=\"nav-item mb-2\" (click)=\"selectedTab = 1\">\r\n            <a\r\n              class=\"nav-link\"\r\n              [ngClass]=\"{ active: selectedTab === 1 }\"\r\n              role=\"tab\"\r\n              href=\"javascript:void(0)\"\r\n              >{{ 'AbpAccount::PersonalSettings' | abpLocalization }}</a\r\n            >\r\n          </li>\r\n        </ul>\r\n      </div>\r\n      @if (profile$ | async) {\r\n        <div class=\"col-12 col-md-9\">\r\n          @if (selectedTab === 0) {\r\n            <div class=\"tab-content\" [@fadeIn]>\r\n              <div class=\"tab-pane active\" role=\"tabpanel\">\r\n                <h4>\r\n                  {{ 'AbpIdentity::ChangePassword' | abpLocalization }}\r\n                  <hr />\r\n                </h4>\r\n                <abp-change-password-form\r\n                  *abpReplaceableTemplate=\"{\r\n                    componentKey: changePasswordKey\r\n                  }\"\r\n                ></abp-change-password-form>\r\n              </div>\r\n            </div>\r\n          }\r\n          @if (selectedTab === 1) {\r\n            <div class=\"tab-content\" [@fadeIn]>\r\n              <div class=\"tab-pane active\" role=\"tabpanel\">\r\n                <h4>\r\n                  {{ 'AbpIdentity::PersonalSettings' | abpLocalization }}\r\n                  <hr />\r\n                </h4>\r\n                <abp-personal-settings-form\r\n                  *abpReplaceableTemplate=\"{\r\n                    componentKey: personalSettingsKey\r\n                  }\"\r\n                ></abp-personal-settings-form>\r\n              </div>\r\n            </div>\r\n          }\r\n        </div>\r\n      }\r\n    </div>\r\n  </div>\r\n</div>\r\n", styles: [".min-h-400{min-height:400px}\n"] }]
        }], ctorParameters: () => [{ type: i2.ProfileService }, { type: ManageProfileStateService }] });

const { maxLength, required, email } = Validators;
class RegisterComponent {
    constructor(fb, accountService, configState, toasterService, authService, injector) {
        this.fb = fb;
        this.accountService = accountService;
        this.configState = configState;
        this.toasterService = toasterService;
        this.authService = authService;
        this.injector = injector;
        this.isSelfRegistrationEnabled = true;
        this.authWrapperKey = "Account.AuthWrapperComponent" /* eAccountComponents.AuthWrapper */;
    }
    ngOnInit() {
        this.init();
        this.buildForm();
    }
    init() {
        this.isSelfRegistrationEnabled =
            (this.configState.getSetting('Abp.Account.IsSelfRegistrationEnabled') || '').toLowerCase() !==
                'false';
        if (!this.isSelfRegistrationEnabled) {
            this.toasterService.warn({
                key: 'AbpAccount::SelfRegistrationDisabledMessage',
                defaultValue: 'Self registration is disabled.',
            }, '', { life: 10000 });
            return;
        }
    }
    buildForm() {
        this.form = this.fb.group({
            username: ['', [required, maxLength(255)]],
            password: ['', [required, ...getPasswordValidators(this.injector)]],
            email: ['', [required, email]],
        });
    }
    onSubmit() {
        if (this.form.invalid)
            return;
        this.inProgress = true;
        const newUser = {
            userName: this.form.get('username')?.value,
            password: this.form.get('password')?.value,
            emailAddress: this.form.get('email')?.value,
            appName: 'Angular',
        };
        this.accountService
            .register(newUser)
            .pipe(switchMap(() => this.authService.login({
            username: newUser.userName,
            password: newUser.password,
            redirectUrl: getRedirectUrl(this.injector),
        })), catchError(err => {
            this.toasterService.error(err.error?.error_description ||
                err.error?.error.message ||
                'AbpAccount::DefaultErrorMessage', '', { life: 7000 });
            return throwError(err);
        }), finalize(() => (this.inProgress = false)))
            .subscribe();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: RegisterComponent, deps: [{ token: i1.UntypedFormBuilder }, { token: i2.AccountService }, { token: i3$1.ConfigStateService }, { token: i4.ToasterService }, { token: i3$1.AuthService }, { token: i0.Injector }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "19.1.8", type: RegisterComponent, isStandalone: false, selector: "abp-register", ngImport: i0, template: "<h4>{{ 'AbpAccount::Register' | abpLocalization }}</h4>\r\n<strong>\r\n  {{ 'AbpAccount::AlreadyRegistered' | abpLocalization }}\r\n  <a class=\"text-decoration-none\" routerLink=\"/account/login\">{{\r\n    'AbpAccount::Login' | abpLocalization\r\n  }}</a>\r\n</strong>\r\n@if (isSelfRegistrationEnabled) {\r\n  <form\r\n    [formGroup]=\"form\"\r\n    (ngSubmit)=\"onSubmit()\"\r\n    validateOnSubmit\r\n    class=\"mt-4\"\r\n  >\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-user-name\" class=\"form-label\">{{\r\n        'AbpAccount::UserName' | abpLocalization\r\n      }}</label\r\n      ><span> * </span\r\n      ><input\r\n        autofocus\r\n        type=\"text\"\r\n        id=\"input-user-name\"\r\n        class=\"form-control\"\r\n        formControlName=\"username\"\r\n        autocomplete=\"username\"\r\n      />\r\n    </div>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-email-address\" class=\"form-label\">{{\r\n        'AbpAccount::EmailAddress' | abpLocalization\r\n      }}</label\r\n      ><span> * </span\r\n      ><input type=\"email\" id=\"input-email-address\" class=\"form-control\" formControlName=\"email\" />\r\n    </div>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-password\" class=\"form-label\">{{\r\n        'AbpAccount::Password' | abpLocalization\r\n      }}</label\r\n      ><span> * </span\r\n      ><input\r\n        type=\"password\"\r\n        id=\"input-password\"\r\n        class=\"form-control\"\r\n        formControlName=\"password\"\r\n        autocomplete=\"current-password\"\r\n      />\r\n    </div>\r\n    <abp-button\r\n      [loading]=\"inProgress\"\r\n      buttonType=\"submit\"\r\n      name=\"Action\"\r\n      buttonClass=\"btn-block btn-lg mt-3 btn btn-primary\"\r\n    >\r\n      {{ 'AbpAccount::Register' | abpLocalization }}\r\n    </abp-button>\r\n  </form>\r\n}\r\n", dependencies: [{ kind: "directive", type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { kind: "directive", type: i1.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { kind: "directive", type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { kind: "directive", type: i1.FormControlName, selector: "[formControlName]", inputs: ["formControlName", "disabled", "ngModel"], outputs: ["ngModelChange"] }, { kind: "directive", type: i3.RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "info", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "directive", type: i3$1.AutofocusDirective, selector: "[autofocus]", inputs: ["autofocus"] }, { kind: "directive", type: i3$1.FormSubmitDirective, selector: "form[ngSubmit][formGroup]", inputs: ["debounce", "notValidateOnSubmit", "markAsDirtyWhenSubmit"], outputs: ["ngSubmit"] }, { kind: "directive", type: i6.ValidationGroupDirective, selector: "[formGroup],[formGroupName]", exportAs: ["validationGroup"] }, { kind: "directive", type: i6.ValidationDirective, selector: "[formControl],[formControlName]", exportAs: ["validationDirective"] }, { kind: "component", type: i4.ButtonComponent, selector: "abp-button", inputs: ["buttonId", "buttonClass", "buttonType", "formName", "iconClass", "loading", "disabled", "attributes"], outputs: ["click", "focus", "blur", "abpClick", "abpFocus", "abpBlur"] }, { kind: "pipe", type: i3$1.LocalizationPipe, name: "abpLocalization" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: RegisterComponent, decorators: [{
            type: Component,
            args: [{ standalone: false, selector: 'abp-register', template: "<h4>{{ 'AbpAccount::Register' | abpLocalization }}</h4>\r\n<strong>\r\n  {{ 'AbpAccount::AlreadyRegistered' | abpLocalization }}\r\n  <a class=\"text-decoration-none\" routerLink=\"/account/login\">{{\r\n    'AbpAccount::Login' | abpLocalization\r\n  }}</a>\r\n</strong>\r\n@if (isSelfRegistrationEnabled) {\r\n  <form\r\n    [formGroup]=\"form\"\r\n    (ngSubmit)=\"onSubmit()\"\r\n    validateOnSubmit\r\n    class=\"mt-4\"\r\n  >\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-user-name\" class=\"form-label\">{{\r\n        'AbpAccount::UserName' | abpLocalization\r\n      }}</label\r\n      ><span> * </span\r\n      ><input\r\n        autofocus\r\n        type=\"text\"\r\n        id=\"input-user-name\"\r\n        class=\"form-control\"\r\n        formControlName=\"username\"\r\n        autocomplete=\"username\"\r\n      />\r\n    </div>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-email-address\" class=\"form-label\">{{\r\n        'AbpAccount::EmailAddress' | abpLocalization\r\n      }}</label\r\n      ><span> * </span\r\n      ><input type=\"email\" id=\"input-email-address\" class=\"form-control\" formControlName=\"email\" />\r\n    </div>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-password\" class=\"form-label\">{{\r\n        'AbpAccount::Password' | abpLocalization\r\n      }}</label\r\n      ><span> * </span\r\n      ><input\r\n        type=\"password\"\r\n        id=\"input-password\"\r\n        class=\"form-control\"\r\n        formControlName=\"password\"\r\n        autocomplete=\"current-password\"\r\n      />\r\n    </div>\r\n    <abp-button\r\n      [loading]=\"inProgress\"\r\n      buttonType=\"submit\"\r\n      name=\"Action\"\r\n      buttonClass=\"btn-block btn-lg mt-3 btn btn-primary\"\r\n    >\r\n      {{ 'AbpAccount::Register' | abpLocalization }}\r\n    </abp-button>\r\n  </form>\r\n}\r\n" }]
        }], ctorParameters: () => [{ type: i1.UntypedFormBuilder }, { type: i2.AccountService }, { type: i3$1.ConfigStateService }, { type: i4.ToasterService }, { type: i3$1.AuthService }, { type: i0.Injector }] });

const PASSWORD_FIELDS = ['password', 'confirmPassword'];
class ResetPasswordComponent {
    constructor(fb, accountService, route, router, injector) {
        this.fb = fb;
        this.accountService = accountService;
        this.route = route;
        this.router = router;
        this.injector = injector;
        this.inProgress = false;
        this.isPasswordReset = false;
        this.mapErrorsFn = (errors, groupErrors, control) => {
            if (PASSWORD_FIELDS.indexOf(String(control?.name)) < 0)
                return errors;
            return errors.concat(groupErrors.filter(({ key }) => key === 'passwordMismatch'));
        };
    }
    ngOnInit() {
        this.route.queryParams.subscribe(({ userId, resetToken }) => {
            if (!userId || !resetToken)
                this.router.navigateByUrl('/account/login');
            this.form = this.fb.group({
                userId: [userId, [Validators.required]],
                resetToken: [resetToken, [Validators.required]],
                password: ['', [Validators.required, ...getPasswordValidators(this.injector)]],
                confirmPassword: ['', [Validators.required, ...getPasswordValidators(this.injector)]],
            }, {
                validators: [comparePasswords(PASSWORD_FIELDS)],
            });
        });
    }
    onSubmit() {
        if (this.form.invalid || this.inProgress)
            return;
        this.inProgress = true;
        this.accountService
            .resetPassword({
            userId: this.form.get('userId')?.value,
            resetToken: this.form.get('resetToken')?.value,
            password: this.form.get('password')?.value,
        })
            .pipe(finalize(() => (this.inProgress = false)))
            .subscribe(() => {
            this.isPasswordReset = true;
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ResetPasswordComponent, deps: [{ token: i1.UntypedFormBuilder }, { token: i2.AccountService }, { token: i3.ActivatedRoute }, { token: i3.Router }, { token: i0.Injector }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "19.1.8", type: ResetPasswordComponent, isStandalone: false, selector: "abp-reset-password", ngImport: i0, template: "<h4>{{ 'AbpAccount::ResetPassword' | abpLocalization }}</h4>\r\n\r\n@if (!isPasswordReset) {\r\n  <form [formGroup]=\"form\" [mapErrorsFn]=\"mapErrorsFn\" (ngSubmit)=\"onSubmit()\" validateOnSubmit>\r\n    <p>{{ 'AbpAccount::ResetPassword_Information' | abpLocalization }}</p>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-password\" class=\"form-label\">{{\r\n        'AbpAccount::Password' | abpLocalization\r\n      }}</label\r\n      ><span> * </span>\r\n      <input type=\"password\" id=\"input-password\" class=\"form-control\" formControlName=\"password\" />\r\n    </div>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-confirm-password\" class=\"form-label\">{{\r\n        'AbpAccount::ConfirmPassword' | abpLocalization\r\n      }}</label\r\n      ><span> * </span>\r\n      <input\r\n        type=\"password\"\r\n        id=\"input-confirm-password\"\r\n        class=\"form-control\"\r\n        formControlName=\"confirmPassword\"\r\n      />\r\n    </div>\r\n    <button class=\"me-2 btn btn-outline-primary\" type=\"button\" routerLink=\"/account/login\">\r\n      {{ 'AbpAccount::Cancel' | abpLocalization }}\r\n    </button>\r\n    <abp-button\r\n      buttonType=\"submit\"\r\n      buttonClass=\"me-2 btn btn-primary\"\r\n      [loading]=\"inProgress\"\r\n      (click)=\"onSubmit()\"\r\n    >\r\n      {{ 'AbpAccount::Submit' | abpLocalization }}\r\n    </abp-button>\r\n  </form>\r\n} @else {\r\n  <p>\r\n    {{ 'AbpAccount::YourPasswordIsSuccessfullyReset' | abpLocalization }}\r\n  </p>\r\n\r\n  <a routerLink=\"/account/login\">\r\n    <button class=\"d-block mt-2 mb-3 btn btn-primary\">\r\n      {{ 'AbpAccount::BackToLogin' | abpLocalization }}\r\n    </button>\r\n  </a>\r\n}\r\n", dependencies: [{ kind: "directive", type: i1.ɵNgNoValidate, selector: "form:not([ngNoForm]):not([ngNativeValidate])" }, { kind: "directive", type: i1.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { kind: "directive", type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { kind: "directive", type: i1.FormControlName, selector: "[formControlName]", inputs: ["formControlName", "disabled", "ngModel"], outputs: ["ngModelChange"] }, { kind: "directive", type: i3.RouterLink, selector: "[routerLink]", inputs: ["target", "queryParams", "fragment", "queryParamsHandling", "state", "info", "relativeTo", "preserveFragment", "skipLocationChange", "replaceUrl", "routerLink"] }, { kind: "directive", type: i3$1.FormSubmitDirective, selector: "form[ngSubmit][formGroup]", inputs: ["debounce", "notValidateOnSubmit", "markAsDirtyWhenSubmit"], outputs: ["ngSubmit"] }, { kind: "directive", type: i6.ValidationGroupDirective, selector: "[formGroup],[formGroupName]", exportAs: ["validationGroup"] }, { kind: "directive", type: i6.ValidationDirective, selector: "[formControl],[formControlName]", exportAs: ["validationDirective"] }, { kind: "component", type: i4.ButtonComponent, selector: "abp-button", inputs: ["buttonId", "buttonClass", "buttonType", "formName", "iconClass", "loading", "disabled", "attributes"], outputs: ["click", "focus", "blur", "abpClick", "abpFocus", "abpBlur"] }, { kind: "pipe", type: i3$1.LocalizationPipe, name: "abpLocalization" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: ResetPasswordComponent, decorators: [{
            type: Component,
            args: [{ standalone: false, selector: 'abp-reset-password', template: "<h4>{{ 'AbpAccount::ResetPassword' | abpLocalization }}</h4>\r\n\r\n@if (!isPasswordReset) {\r\n  <form [formGroup]=\"form\" [mapErrorsFn]=\"mapErrorsFn\" (ngSubmit)=\"onSubmit()\" validateOnSubmit>\r\n    <p>{{ 'AbpAccount::ResetPassword_Information' | abpLocalization }}</p>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-password\" class=\"form-label\">{{\r\n        'AbpAccount::Password' | abpLocalization\r\n      }}</label\r\n      ><span> * </span>\r\n      <input type=\"password\" id=\"input-password\" class=\"form-control\" formControlName=\"password\" />\r\n    </div>\r\n    <div class=\"mb-3 form-group\">\r\n      <label for=\"input-confirm-password\" class=\"form-label\">{{\r\n        'AbpAccount::ConfirmPassword' | abpLocalization\r\n      }}</label\r\n      ><span> * </span>\r\n      <input\r\n        type=\"password\"\r\n        id=\"input-confirm-password\"\r\n        class=\"form-control\"\r\n        formControlName=\"confirmPassword\"\r\n      />\r\n    </div>\r\n    <button class=\"me-2 btn btn-outline-primary\" type=\"button\" routerLink=\"/account/login\">\r\n      {{ 'AbpAccount::Cancel' | abpLocalization }}\r\n    </button>\r\n    <abp-button\r\n      buttonType=\"submit\"\r\n      buttonClass=\"me-2 btn btn-primary\"\r\n      [loading]=\"inProgress\"\r\n      (click)=\"onSubmit()\"\r\n    >\r\n      {{ 'AbpAccount::Submit' | abpLocalization }}\r\n    </abp-button>\r\n  </form>\r\n} @else {\r\n  <p>\r\n    {{ 'AbpAccount::YourPasswordIsSuccessfullyReset' | abpLocalization }}\r\n  </p>\r\n\r\n  <a routerLink=\"/account/login\">\r\n    <button class=\"d-block mt-2 mb-3 btn btn-primary\">\r\n      {{ 'AbpAccount::BackToLogin' | abpLocalization }}\r\n    </button>\r\n  </a>\r\n}\r\n" }]
        }], ctorParameters: () => [{ type: i1.UntypedFormBuilder }, { type: i2.AccountService }, { type: i3.ActivatedRoute }, { type: i3.Router }, { type: i0.Injector }] });

/**
 * @deprecated Use `authenticationFlowGuard` *function* instead.
 */
class AuthenticationFlowGuard {
    constructor() {
        this.authService = inject(AuthService);
    }
    canActivate() {
        if (this.authService.isInternalAuth)
            return true;
        this.authService.navigateToLogin();
        return false;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AuthenticationFlowGuard, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AuthenticationFlowGuard }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AuthenticationFlowGuard, decorators: [{
            type: Injectable
        }] });
const authenticationFlowGuard = () => {
    const authService = inject(AuthService);
    if (authService.isInternalAuth)
        return true;
    authService.navigateToLogin();
    return false;
};

/**
 * @deprecated Use `accountExtensionsResolver` *function* instead.
 */
class AccountExtensionsGuard {
    constructor() {
        this.injector = inject(Injector);
        this.extensions = inject(ExtensionsService);
    }
    canActivate() {
        const config = { optional: true };
        const editFormContributors = inject(ACCOUNT_EDIT_FORM_PROP_CONTRIBUTORS, config) || {};
        return getObjectExtensionEntitiesFromStore(this.injector, 'Identity').pipe(map(entities => ({
            ["Account.PersonalSettingsComponent" /* eAccountComponents.PersonalSettings */]: entities.User,
        })), mapEntitiesToContributors(this.injector, 'AbpIdentity'), tap(objectExtensionContributors => {
            mergeWithDefaultProps(this.extensions.editFormProps, DEFAULT_ACCOUNT_FORM_PROPS, objectExtensionContributors.editForm, editFormContributors);
        }), map(() => true));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AccountExtensionsGuard, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AccountExtensionsGuard }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AccountExtensionsGuard, decorators: [{
            type: Injectable
        }] });

const accountExtensionsResolver = () => {
    const injector = inject(Injector);
    const extensions = inject(ExtensionsService);
    const config = { optional: true };
    const editFormContributors = inject(ACCOUNT_EDIT_FORM_PROP_CONTRIBUTORS, config) || {};
    return getObjectExtensionEntitiesFromStore(injector, 'Identity').pipe(map$1(entities => ({
        ["Account.PersonalSettingsComponent" /* eAccountComponents.PersonalSettings */]: entities.User,
    })), mapEntitiesToContributors(injector, 'AbpIdentity'), tap$1(objectExtensionContributors => {
        mergeWithDefaultProps(extensions.editFormProps, DEFAULT_ACCOUNT_FORM_PROPS, objectExtensionContributors.editForm, editFormContributors);
    }));
};

const canActivate = [authenticationFlowGuard];
const routes = [
    { path: '', pathMatch: 'full', redirectTo: 'login' },
    {
        path: '',
        component: RouterOutletComponent,
        children: [
            {
                path: 'login',
                component: ReplaceableRouteContainerComponent,
                canActivate,
                data: {
                    replaceableComponent: {
                        key: "Account.LoginComponent" /* eAccountComponents.Login */,
                        defaultComponent: LoginComponent,
                    },
                },
                title: 'AbpAccount::Login',
            },
            {
                path: 'register',
                component: ReplaceableRouteContainerComponent,
                canActivate,
                data: {
                    replaceableComponent: {
                        key: "Account.RegisterComponent" /* eAccountComponents.Register */,
                        defaultComponent: RegisterComponent,
                    },
                },
                title: 'AbpAccount::Register',
            },
            {
                path: 'forgot-password',
                component: ReplaceableRouteContainerComponent,
                canActivate,
                data: {
                    replaceableComponent: {
                        key: "Account.ForgotPasswordComponent" /* eAccountComponents.ForgotPassword */,
                        defaultComponent: ForgotPasswordComponent,
                    },
                },
                title: 'AbpAccount::ForgotPassword',
            },
            {
                path: 'reset-password',
                component: ReplaceableRouteContainerComponent,
                canActivate: [],
                data: {
                    tenantBoxVisible: false,
                    replaceableComponent: {
                        key: "Account.ResetPasswordComponent" /* eAccountComponents.ResetPassword */,
                        defaultComponent: ResetPasswordComponent,
                    },
                },
                title: 'AbpAccount::ResetPassword',
            },
            {
                path: 'manage',
                component: ReplaceableRouteContainerComponent,
                canActivate: [authGuard],
                resolve: [accountExtensionsResolver],
                data: {
                    replaceableComponent: {
                        key: "Account.ManageProfileComponent" /* eAccountComponents.ManageProfile */,
                        defaultComponent: ManageProfileComponent,
                    },
                },
                title: 'AbpAccount::MyAccount',
            },
        ],
    },
];
class AccountRoutingModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AccountRoutingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "19.1.8", ngImport: i0, type: AccountRoutingModule, imports: [i3.RouterModule], exports: [RouterModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AccountRoutingModule, imports: [RouterModule.forChild(routes), RouterModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AccountRoutingModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [RouterModule.forChild(routes)],
                    exports: [RouterModule],
                }]
        }] });

function accountConfigOptionsFactory(options) {
    return {
        redirectUrl: '/',
        ...options,
    };
}

const declarations = [
    LoginComponent,
    RegisterComponent,
    ChangePasswordComponent,
    ManageProfileComponent,
    PersonalSettingsComponent,
    ForgotPasswordComponent,
    ResetPasswordComponent,
    PersonalSettingsHalfRowComponent,
];
class AccountModule {
    static forChild(options = {}) {
        return {
            ngModule: AccountModule,
            providers: [
                AuthenticationFlowGuard,
                { provide: ACCOUNT_CONFIG_OPTIONS, useValue: options },
                {
                    provide: 'ACCOUNT_OPTIONS',
                    useFactory: accountConfigOptionsFactory,
                    deps: [ACCOUNT_CONFIG_OPTIONS],
                },
                {
                    provide: RE_LOGIN_CONFIRMATION_TOKEN,
                    useValue: options.isPersonalSettingsChangedConfirmationActive ?? true,
                },
                {
                    provide: ACCOUNT_EDIT_FORM_PROP_CONTRIBUTORS,
                    useValue: options.editFormPropContributors,
                },
                AccountExtensionsGuard,
            ],
        };
    }
    static forLazy(options = {}) {
        return new LazyModuleFactory(AccountModule.forChild(options));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AccountModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "19.1.8", ngImport: i0, type: AccountModule, declarations: [LoginComponent,
            RegisterComponent,
            ChangePasswordComponent,
            ManageProfileComponent,
            PersonalSettingsComponent,
            ForgotPasswordComponent,
            ResetPasswordComponent,
            PersonalSettingsHalfRowComponent], imports: [CoreModule,
            AccountRoutingModule,
            ThemeSharedModule,
            NgbDropdownModule,
            NgxValidateCoreModule,
            ExtensibleModule], exports: [LoginComponent,
            RegisterComponent,
            ChangePasswordComponent,
            ManageProfileComponent,
            PersonalSettingsComponent,
            ForgotPasswordComponent,
            ResetPasswordComponent,
            PersonalSettingsHalfRowComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AccountModule, imports: [CoreModule,
            AccountRoutingModule,
            ThemeSharedModule,
            NgbDropdownModule,
            NgxValidateCoreModule,
            ExtensibleModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "19.1.8", ngImport: i0, type: AccountModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [...declarations],
                    imports: [
                        CoreModule,
                        AccountRoutingModule,
                        ThemeSharedModule,
                        NgbDropdownModule,
                        NgxValidateCoreModule,
                        ExtensibleModule,
                    ],
                    exports: [...declarations],
                }]
        }] });

/* eslint-disable @typescript-eslint/no-empty-interface */

/**
 * Generated bundle index. Do not edit.
 */

export { ACCOUNT_CONFIG_OPTIONS, ACCOUNT_EDIT_FORM_PROP_CONTRIBUTORS, AccountExtensionsGuard, AccountModule, AuthenticationFlowGuard, ChangePasswordComponent, DEFAULT_ACCOUNT_FORM_PROPS, ForgotPasswordComponent, LoginComponent, ManageProfileComponent, ManageProfileStateService, PersonalSettingsComponent, PersonalSettingsHalfRowComponent, RE_LOGIN_CONFIRMATION_TOKEN, RegisterComponent, ResetPasswordComponent, accountConfigOptionsFactory, accountExtensionsResolver, authenticationFlowGuard, getRedirectUrl };
